components/sceneutil/riggeometry.cpp

```diff
@@ -141,7 +141,7 @@ bool RigGeometry::initFromParentSkeleton(osg::NodeVisitor* nv)
     return true;
 }
 
-void accumulateMatrix(const osg::Matrixf& invBindMatrix, const osg::Matrixf& matrix, float weight, osg::Matrixf& result)
+inline void accumulateMatrix(const osg::Matrixf& invBindMatrix, const osg::Matrixf& matrix, const float& weight, osg::Matrixf& result)
```

Comment:
Better put this function into anonymous namespace. C++ standard does not say how to make an assembly code with `call` or without for function with `inline` specifier. But for optimizers anonymous namespace tells that this function will not be accessed from any other translation unit. This function is declared and defined only in this file, so to avoid any name conflicts anonymous namespace is more usefull. You can try to use [compiler explorer](https://github.com/mattgodbolt/compiler-explorer) to check compiler assembly output and compare it for different approaches (for openmw only local setup can be used). However there is no difference between `inline` and anonymous namespace for clang 6 and gcc 8 output, so right now there is no any problem in this code.

Reference to float doesn't have any effect if function is already inlined. But for not inlined it leads to one additional `push` to save one more register. ![diff](https://user-images.githubusercontent.com/764107/44620372-b7b09380-a89b-11e8-9e41-308f6e1a6c13.png)

Reply:
Done.